phase1-coordinator/src/coordinator_state.rs

```diff
@@ -1514,6 +1526,20 @@ impl CoordinatorState {
             }
         }
 
+        // Zero the reliability score if the participant is joining with a known IP.
+        if let Some(ip) = participant_ip {
+            if self.is_duplicate_ip(&ip) {
+                reliability_score = 0;
+
+                // Also zero the reliability scores of existing participants in the queue with the
+                // same IP.
+                self.zero_duplicate_ips(&ip);
```

Comment:
not that familiar with the inner workings here, but I'm guessing the participant remains in the queue only so that they can be used in case they are the only one left when the end of the queue is reached?

Reply:
Yes. Also the assumption is these are people running bots/have multiple tabs open on the browser setup. We want to prioritise honest contributors, hence moving the spammers to the back, but if there are none of the former in the queue it's fine if later contribute.